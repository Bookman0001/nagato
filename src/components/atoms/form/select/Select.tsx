import React, { SelectHTMLAttributes } from 'react'
import styled from 'styled-components'

import { borderRadius, color } from 'src/theme/constants'

interface Props extends SelectHTMLAttributes<HTMLSelectElement> {
  children: React.ReactChild
}

export function Select({
  onChange = () => null,
  defaultValue = '',
  ...props
}: Props) {
  return (
    <Container>
      <StyledSelect defaultValue={defaultValue} onChange={onChange} {...props}>
        <>{props.children}</>
      </StyledSelect>
    </Container>
  )
}

const Container = styled.div`
  position: relative;
  border: 1px solid ${color.BLACK};
  border-radius: ${borderRadius.DEFAULT};
  background: ${color.WHITE};
  line-height: 1.5;
  &::before {
    position: absolute;
    top: 0.6em;
    right: 0.6em;
    width: 0;
    height: 0;
    padding: 0;
    content: '';
    border-left: 6px solid transparent;
    border-right: 6px solid transparent;
    border-top: 6px solid ${color.BLACK};
    pointer-events: none;
  }
`

const StyledSelect = styled.select`
  color: ${color.BLACK};
  width: 100%;
  height: 100%;
  padding: 0 1em;
  cursor: pointer;
  text-indent: 0.01px;
  text-overflow: ellipsis;
  border: none;
  outline: none;
  background: transparent;
  background-image: none;
  box-shadow: none;
  -webkit-appearance: none;
  appearance: none;
`
