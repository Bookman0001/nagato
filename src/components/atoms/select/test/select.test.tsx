import { shallow } from 'enzyme'

import Select from '../'

test('SearchInput', () => {
  const component = shallow(
    <Select defaultValue={''} onChange={(e) => console.log(e.target.value)}>
      <>
        <option>Option1</option>
      </>
    </Select>
  )
  expect(component).toMatchSnapshot()
})
