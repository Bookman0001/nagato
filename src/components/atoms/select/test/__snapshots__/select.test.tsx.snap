// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput 1`] = `
<styled.div>
  <styled.select
    defaultValue=""
    onChange={[Function]}
  >
    <option>
      Option1
    </option>
  </styled.select>
</styled.div>
`;
